seshdash/utils/alert.py

```diff
@@ -13,46 +15,81 @@
 # Alert_Rule.objects.create(site = site, check_field="battery_voltage", value=25, operator="lt")
 
 # Rules are based on sites. So you need to define a rule for each site if you want to check same configurations in several sites.
-# A Sesh_Alert object is created for each alert triggered and an email is send if the rule has send_mail option true
-def alert_check(data_point):
+# A Sesh_Alert object is created for each 'alert triggered and an email is send if the rule has send_mail option true
+def alert_check(site):
     ops = {'lt': lambda x,y: x<y,
            'gt': lambda x,y: x>y,
            'eq' : lambda x,y: x==y,
     }
-    recipients = []
+    email_recipients = []
+    sms_recipients = []
     content = {}
 
-    rules = Alert_Rule.objects.filter(site = data_point.site)
+    rules = Alert_Rule.objects.filter(site = site)
     for rule in rules:
-        real_value = getattr(data_point,rule.check_field)
+
+        if '#' in rule.check_field: # If the rule is valid (contains model and field_name)
+            model, field_name = rule.check_field.split('#')
+             
+            # Getting the model name and the latest value of the model field
+            model = get_model_from_string(model)
+            data_point = model.objects.all().order_by('-id')[0]
+            real_value = getattr(data_point, field_name)
+
         if ops[rule.operator](real_value,rule.value):
-            content_str = "site:%s\nrule:%s '%s' %s --> found %s " %(data_point.site.site_name,rule.check_field,rule.operator,rule.value,real_value)
+            content_str = "site:%s\nrule:%s '%s' %s --> found %s " %(site.site_name,rule.check_field,rule.operator,rule.value,real_value)
 
             # Get ready content for email
-            content['site'] = data_point.site.site_name
+            content['site'] = site.site_name
             content['alert'] = content_str
             content['time'] = data_point.time
             content['data_point'] = data_point
 
             # TODO rule object should have the list of related persons to send alert mail
-            users = get_users_with_perms(data_point.site)
+            users = get_users_with_perms(site)
+            # TODO to be removed
+            userSms = Sesh_User.objects.all()
```

Comment:
why are you referencing users and userSMS separately. Can't you get one set from the other? I think you can use the relationship here? Or do I have to fix the relationship first?